commit 41bc5e09352ca838e1787f3109dcf4b4ae85986a
Author: Will Newton <will.newton@linaro.org>
Date:   Wed Sep 25 10:17:28 2013 +0100

    ARM: Allow building __sigsetjmp as Thumb.
    
    Convert __sigsetjmp code to allow building as Thumb.
    
    ports/ChangeLog.arm:
    
    2013-10-04  Will Newton  <will.newton@linaro.org>
    
    	* sysdeps/arm/setjmp.S (NO_THUMB): Remove define.
    	(__sigsetjmp): Use Thumb supported instructions.

diff --git a/ports/sysdeps/arm/setjmp.S b/ports/sysdeps/arm/setjmp.S
index b38b919..fedd994 100644
--- a/ports/sysdeps/arm/setjmp.S
+++ b/ports/sysdeps/arm/setjmp.S
@@ -16,8 +16,6 @@
    License along with the GNU C Library.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
-/* ??? Needs more rearrangement for the STM to handle thumb mode.  */
-#define NO_THUMB
 #include <sysdep.h>
 #include <bits/setjmp.h>
 #include <rtld-global-offsets.h>
@@ -35,7 +33,8 @@ ENTRY (__sigsetjmp)
 	sfi_breg ip, \
 	stmia	\B!, JMP_BUF_REGLIST
 #ifdef PTR_MANGLE
-	PTR_MANGLE2 (a4, sp, a3)
+	mov	a4, sp
+	PTR_MANGLE2 (a4, a4, a3)
 	str	a4, [ip], #4
 	PTR_MANGLE2 (a4, lr, a3)
 	str	a4, [ip], #4
